tests/core/test_visualization.py

```diff
@@ -90,12 +90,13 @@ def test_common_action_prefix_unequal():
         should_merge_nodes=False,
     )
 
-    generated_graph = nx_pydot.to_pydot(generated_graph)
+    generated_graph = str(nx_pydot.to_pydot(generated_graph))
 
     assert isfile(out_file)
 
     with open(out_file, "r") as graph_file:
         content = graph_file.read()
 
+    assert r'label="/average_weather\{\"season\":\ \"winter\"\}"' in str(generated_graph)
```

Comment:
you did that already above
```suggestion
    assert r'label="/average_weather\{\"season\":\ \"winter\"\}"' in generated_graph
```

Reply:
Yeah I originally made this two tests and left that when combining them